packages/primitives/src/reatomSet.ts

```diff
@@ -43,6 +51,36 @@ export const reatomSet = <T>(
         })
       }, `${name}.clear`),
       reset: action((ctx) => target(ctx, initState), `${name}.reset`),
+      intersection: action(
+        (ctx, set) => target(ctx, (prev) => new Set(prev).intersection(set)),
```

Comment:
These new methods already creates a new set, so you don't need to do it by yourself

Reply:
done

Reply:
![Uploading CleanShot 2024-06-09 at 11.12.09@2x.png…]()